### PR TITLE
Expose gesture recognizers directly on MapView.GestureManager

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Gesutre Recognizers added to the mapView are exposed as properties on `mapView.gestures`</changelog>`.
 - [ ] Update the migration guide, API Docs, Markdown files - Readme, Developing, etc

### Summary of changes

This PR adds computed vars to the public interface of `MapView.GestureManager` which allow for a developer to inspect (and potentially become the `GestureRecognizerDelegate` of) stock gestures that are set on the map.

### CODE REVIEW

File: Sources/MapboxMaps/Gestures/GestureManager.swift [hunk 1]
```diff
@@ -19,6 +19,80 @@ public final class GestureManager: NSObject {
...
     private(set) var gestureHandlers: [GestureType: GestureHandler] = [:]
 
+    /// The underlying gesture recognizer for the pan gesture
+    public var panGestureRecognizer: UIPanGestureRecognizer? {
+        if let handler = gestureHandlers[.pan],
+           let recognizer = handler.gestureRecognizer as? UIPanGestureRecognizer {
+            return recognizer
+        }
+
+        return nil
```
Comment (macdrevx) [inline Sources/MapboxMaps/Gestures/GestureManager.swift:29]:
```suggestion
        return gestureHandlers[.pan]?.gestureRecognizer as? UIPanGestureRecognizer
```

Does that work?

**Comment by macdrevx** [comment]
What do you think of making the property types simply `UIGestureRecognizer` that way we're less tied to a specific implementation?

**Comment by nishant-karajgikar** [comment]
> What do you think of making the property types simply `UIGestureRecognizer` that way we're less tied to a specific implementation?

Yep, I'm not opposed to that. Will update.

File: Sources/MapboxMaps/Gestures/GestureManager.swift (hunk 1 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Gestures/GestureManager.swift]: yep, cleaned up!

**Review by julianrex — APPROVED** [review]

**Review by macdrevx — APPROVED** [review]

**Pull request merged**